### PR TITLE
feat(fxa-settings): upload and edit avatar

### LINKED ISSUE #5021: [avatar] Transfer "Upload photo" functionality over
Trigger the file system picker and accept valid photos. On success, send the user to the next screen, however the cropping etc. of the next screen will be handled in another issue. Display errors on this screen in the global error message component.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-1697)
┆Issue Number: FXA-1697

### LINKED ISSUE #5022: [avatar] Upload photo position/rotate/zoom your picture page
After the user has uploaded a photo they're taken to a page where they can adjust the image (zoom, drag, rotate). "Cancel" should take the user back to the previous screen and "Save" should save the photo and display a success message at the top of the Settings screen, functionality should have been built out with FXA-1702 (or can be a reference for processing the response).

This issue is an 5 because of amount of work it will be to transfer over functionality from `fxa-content-server`'s `avatar.js`, `cropper.js`, and related files. It might be worth installing a package to handle a lot of this for us.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-1698)
┆Issue Number: FXA-1698

### PR DESCRIPTION
**Pull request by meandavejustice**
## Because

- We want to allow users to upload their own avatars and edit the images
- Parity with old settings

## This pull request

- Adds upload and image editing capabilities for the add avatar section

## Issue that this pull request solves

- fixes #5021
- fixes #5022

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

image
image

### CODE REVIEW

File: packages/fxa-settings/package.json [hunk 1]
```diff
@@ -33,21 +33,25 @@
   },
   "dependencies": {
     "@apollo/client": "^3.3.7",
+    "@emotion/react": "^11.1.4",
```
Comment (dannycoates) [inline packages/fxa-settings/package.json:36]: ???

File: packages/fxa-settings/src/components/PageAvatar/Add.tsx [hunk 2]
```diff
@@ -2,34 +2,94 @@
...
-  // const account = useAccount();
+  const { l10n } = useLocalization();
+  const { avatarUrl } = useAccount();
+  const fileInputRef = useRef(null);
   const alertBar = useAlertBar();
-  // const [errorText, setErrorText] = useState<string>();
   // const [saveEnabled, setSaveEnabled] = useState<boolean>();
+  const [imageSrc, setImageSrc] = useState(undefined);
   const handleSubmit = async (ev: React.FormEvent<HTMLFormElement>) => {
     ev.preventDefault();
```
Comment (dannycoates) [inline packages/fxa-settings/src/components/PageAvatar/Add.tsx:38]: in the upload success response handler please add a line like this https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/components/PageDisplayName/index.tsx#L48 to notify the browser.

**Review by dannycoates — COMMENTED** [review]
This looks good so far. 👍 Can you mark it as Draft until it's complete?

**Comment by meandavejustice** [comment]
> Can you mark it as Draft until it's complete?

@dannycoates Switched. Is this what we are doing with prs now? until they are approved? I haven't seen this.

File: packages/fxa-settings/package.json (hunk 1 above)
Comment (meandavejustice) [reply packages/fxa-settings/package.json:36]: `@emotion/react` is supposed to be a devDependency for `@material-ui/core/slider`. But I wasn't able to get it working without installing it also. I'm not sure if there is another work around for this.

File: packages/fxa-settings/src/components/PageAvatar/Add.tsx (hunk 2 above)
Comment (meandavejustice) [reply packages/fxa-settings/src/components/PageAvatar/Add.tsx:38]: I have this added in another branch for handling uploads, https://jira.mozilla.com/browse/FXA-1702

**Comment by dannycoates** [comment]
> Have we changed the process for pr reviews?

ah, ok. I thought this PR was going to be for all of the avatar issues. Also the commented out code made me think it wasn't ready.

**Review by dannycoates — APPROVED** [review]
nit to rm the commented code in AvatarCropper. LGTM

**Pull request merged**